packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart

```diff
@@ -498,26 +494,21 @@ class _ResidentWebRunner extends ResidentWebRunner {
       }
     }
 
-    Duration transferMarker;
     try {
       if (!deviceIsDebuggable) {
         globals.printStatus('Recompile complete. Page requires refresh.');
-      } else if (!debuggingOptions.buildInfo.isDebug) {
+      } else if (isRunningDebug) {
+        await _vmService.callMethod('hotRestart');
+      } else {
         // On non-debug builds, a hard refresh is required to ensure the
         // up to date sources are loaded.
         await _wipConnection?.sendCommand('Page.reload', <String, Object>{
           'ignoreCache': !debuggingOptions.buildInfo.isDebug,
         });
-      } else {
-        transferMarker = timer.elapsed;
-        await _wipConnection?.debugger?.sendCommand(
-          'Runtime.evaluate', params: <String, Object>{
-            'expression': 'window.\$hotReloadHook([$reloadModules])',
-            'awaitPromise': true,
-            'returnByValue': true,
-          },
-        );
       }
+    } on Exception catch (err) {
+      globals.printError(err.toString());
```

Comment:
If we can't give any more information here, and it will fail later regardless, should this just be a `printTrace()` ?

Reply:
Commented below